lib/cmsis_rtos_v1/cmsis_signal.c

```diff
@@ -79,7 +81,7 @@ osEvent osSignalWait(int32_t signals, uint32_t millisec)
 	}
 
 	/* Check if signals is within the permitted range */
-	if (signals >= (1 << (osFeature_Signals + 1))) {
+	if ((!signals) || (signals >= (1 << (osFeature_Signals + 1)))) {
```

Comment:
signals = 0 is a valid case. Undo this change.

Reply:
Sorry, my bad, missed reading that part. Code changed accordingly.